onnx/optimizer/passes/optimize_pass.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include "onnx/onnx_pb.h"
+#include "onnx/ir.h"
+
+namespace onnx { namespace optimization {
+
+enum class API_TYPE {
+  proto, ir
+};
+
+struct OptimizePass {
+
+  std::string name;
+  API_TYPE type;
+
+  explicit OptimizePass(const std::string name, API_TYPE type)
+    : name(name), type(type) {
+  }
+
+  virtual void optimize(onnx::ModelProto& mp) {}
+
+  virtual void optimize(std::shared_ptr<Graph>& graph) {}
```

Comment:
why share_ptr?

Reply:
and also why non-const reference to a shared_ptr (it's usually very weird)

Reply:
Good catch, I will fix this.